[Bug Report] Failure during pip install

<!-- Please read carefully before submitting. -->
<!-- Please search to see if an issue already exists for the bug you encountered. -->
<!-- Any text contained in brackets like this will not appear in your final bug report. -->
<!-- Click on the preview tab to see what your report will look like before submitting. -->

## Bug Report
<!-- 
Please report one issue per report.
If you have multiple issues then open another report.
-->
### Current Behaviour:
On Ubuntu 22.04 any 0.10* version fails to build. Can't use any of the prior builds due to the UI bug.

### Steps To Reproduce:
Created a virtual python environment. 
While installing version 0.10.0a1 via pip or any of the newer versions from source the install fails while building the amulet-map-editor wheel.

```
$ python -m pip install /home/josh/.amulet/Amulet-Map-Editor-0.10.0b1 --upgrade

Processing ./.amulet/Amulet-Map-Editor-0.10.0b1
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting amulet-nbt~=2.0
  Using cached amulet_nbt-2.0.1-cp310-cp310-linux_x86_64.whl
Requirement already satisfied: numpy~=1.17 in ./.amulet/lib/python3.10/site-packages (from amulet-map-editor==0.10.0b1) (1.23.1)
Requirement already satisfied: wxPython~=4.1 in ./.amulet/lib/python3.10/site-packages (from amulet-map-editor==0.10.0b1) (4.2.0)
Collecting pyopengl~=3.0
  Using cached PyOpenGL-3.1.6-py3-none-any.whl (2.4 MB)
Collecting packaging
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting amulet-core~=1.9
  Using cached amulet_core-1.9.4-py3-none-any.whl (1.0 MB)
Collecting minecraft-resource-pack~=1.3
  Using cached minecraft_resource_pack-1.3.1-py3-none-any.whl (448 kB)
Collecting pymctranslate~=1.2
  Using cached PyMCTranslate-1.2.6-py3-none-any.whl (701 kB)
Collecting portalocker~=2.4
  Using cached portalocker-2.5.1-py2.py3-none-any.whl (15 kB)
Collecting mutf8~=1.0
  Using cached mutf8-1.0.6-cp310-cp310-linux_x86_64.whl
Requirement already satisfied: Pillow>=7.1 in ./.amulet/lib/python3.10/site-packages (from minecraft-resource-pack~=1.3->amulet-map-editor==0.10.0b1) (9.2.0)
Requirement already satisfied: six in ./.amulet/lib/python3.10/site-packages (from wxPython~=4.1->amulet-map-editor==0.10.0b1) (1.16.0)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Building wheels for collected packages: amulet-map-editor
  Building wheel for amulet-map-editor (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for amulet-map-editor (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [323 lines of output]
      Warning: build in /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages is using versioneer.py from /home/josh/.amulet/Amulet-Map-Editor-0.10.0b1/versioneer.py
      running bdist_wheel
      running build
      running build_py
      Warning: build in /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages is using versioneer.py from /home/josh/.amulet/Amulet-Map-Editor-0.10.0b1/versioneer.py
      Warning: build in /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages is using versioneer.py from /home/josh/.amulet/Amulet-Map-Editor-0.10.0b1/versioneer.py
      running egg_info
      writing amulet_map_editor.egg-info/PKG-INFO
      writing dependency_links to amulet_map_editor.egg-info/dependency_links.txt
      writing entry points to amulet_map_editor.egg-info/entry_points.txt
      writing requirements to amulet_map_editor.egg-info/requires.txt
      writing top-level names to amulet_map_editor.egg-info/top_level.txt
      reading manifest file 'amulet_map_editor.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.pyi' under directory 'amulet_map_editor'
      warning: no files found matching '*.pxd' under directory 'amulet_map_editor'
      warning: no files found matching '*.jpg' under directory 'amulet_map_editor'
      writing manifest file 'amulet_map_editor.egg-info/SOURCES.txt'
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.api.image.icon.tablericons' as data is deprecated, please list it in `packages`.
          !!

          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.api.image.icon.tablericons' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.api.image.icon.tablericons' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.api.image.icon.tablericons' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.api.image.logo' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.api.image.logo' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.api.image.logo' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.api.image.logo' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.api.image.nbt' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.api.image.nbt' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.api.image.nbt' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.api.image.nbt' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.lang' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.lang' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.lang' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.lang' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.amulet_ui' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.amulet_ui' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.amulet_ui' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.amulet_ui' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.amulet_ui.cubemap' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.amulet_ui.cubemap' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.amulet_ui.cubemap' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.amulet_ui.cubemap' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.textures' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.textures' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.textures' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.programs.edit.amulet_resource_pack.bedrock.textures' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.programs.edit.amulet_resource_pack.java' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.programs.edit.amulet_resource_pack.java' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.programs.edit.amulet_resource_pack.java' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.programs.edit.amulet_resource_pack.java' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.programs.edit.amulet_resource_pack.java.assets.amulet.textures.amulet_ui' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.programs.edit.amulet_resource_pack.java.assets.amulet.textures.amulet_ui' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.programs.edit.amulet_resource_pack.java.assets.amulet.textures.amulet_ui' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.programs.edit.amulet_resource_pack.java.assets.amulet.textures.amulet_ui' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'amulet_map_editor.programs.edit.amulet_resource_pack.java.assets.amulet.textures.amulet_ui.cubemap' as data is deprecated, please list it in `packages`.
          !!
      
      
          ############################
          # Package would be ignored #
          ############################
          Python recognizes 'amulet_map_editor.programs.edit.amulet_resource_pack.java.assets.amulet.textures.amulet_ui.cubemap' as an importable package,
          but it is not listed in the `packages` configuration of setuptools.
      
          'amulet_map_editor.programs.edit.amulet_resource_pack.java.assets.amulet.textures.amulet_ui.cubemap' has been automatically added to the distribution only
          because it may contain data files, but this behavior is likely to change
          in future versions of setuptools (and therefore is considered deprecated).
      
          Please make sure that 'amulet_map_editor.programs.edit.amulet_resource_pack.java.assets.amulet.textures.amulet_ui.cubemap' is included as a package by using
          the `packages` configuration field or the proper discovery methods
          (for example by using `find_namespace_packages(...)`/`find_namespace:`
          instead of `find_packages(...)`/`find:`).
      
          You can read more about "package discovery" and "data files" on setuptools
          documentation page.
      
      
      !!
      
        check.warn(importable)
      copying amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.c -> build/lib.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk
      UPDATING build/lib.linux-x86_64-cpython-310/amulet_map_editor/_version.py
      set build/lib.linux-x86_64-cpython-310/amulet_map_editor/_version.py to '0.10.0b1'
      running build_ext
      Warning: build in /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages is using versioneer.py from /home/josh/.amulet/Amulet-Map-Editor-0.10.0b1/versioneer.py
      Warning: build in /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages is using versioneer.py from /home/josh/.amulet/Amulet-Map-Editor-0.10.0b1/versioneer.py
      building 'amulet_map_editor.api.opengl.mesh.level.chunk.chunk_builder_cy' extension
      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include -I/home/josh/.amulet/include -I/usr/include/python3.10 -c amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.c -o build/temp.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.o -openmp
      In file included from /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarraytypes.h:1948,
                       from /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                       from /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/arrayobject.h:5,
                       from amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.c:1071:
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
         17 | #warning "Using deprecated NumPy API, disable it with " \
            |  ^~~~~~~
      x86_64-linux-gnu-gcc -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -g -fwrapv -O2 build/temp.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.o -L/usr/lib/x86_64-linux-gnu -o build/lib.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.cpython-310-x86_64-linux-gnu.so -openmp
      /usr/bin/ld: cannot find build/temp.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.o: No such file or directory
      collect2: error: ld returned 1 exit status
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for amulet-map-editor
Failed to build amulet-map-editor
ERROR: Could not build wheels for amulet-map-editor, which is required to install pyproject.toml-based projects
```


### Environment:
<!-- Please add the details of your environment. -->
- OS: Ubuntu 22.04
- Amulet Version: 0.10.0a1 or greater

I am not entirely sure what the issue is here.
These four lines are where the error actually is
```
x86_64-linux-gnu-gcc -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -g -fwrapv -O2 build/temp.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.o -L/usr/lib/x86_64-linux-gnu -o build/lib.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.cpython-310-x86_64-linux-gnu.so -openmp
/usr/bin/ld: cannot find build/temp.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.o: No such file or directory
collect2: error: ld returned 1 exit status
error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
```

The file is a .pyx file which is a cython file which is converted to a .c file which is compiled to a .so file on linux.
I have just spotted the line further up that should generate the .o file

```
      building 'amulet_map_editor.api.opengl.mesh.level.chunk.chunk_builder_cy' extension
      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include -I/home/josh/.amulet/include -I/usr/include/python3.10 -c amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.c -o build/temp.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.o -openmp
      In file included from /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarraytypes.h:1948,
                       from /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/ndarrayobject.h:12,
                       from /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/arrayobject.h:5,
                       from amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.c:1071:
      /tmp/pip-build-env-unn6l7fk/overlay/lib/python3.10/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: #warning "Using deprecated NumPy API, disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
         17 | #warning "Using deprecated NumPy API, disable it with " \
            |  ^~~~~~~
      x86_64-linux-gnu-gcc -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -g -fwrapv -O2 build/temp.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.o -L/usr/lib/x86_64-linux-gnu -o build/lib.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.cpython-310-x86_64-linux-gnu.so -openmp
      /usr/bin/ld: cannot find build/temp.linux-x86_64-cpython-310/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.o: No such file or directory
      collect2: error: ld returned 1 exit status
```

--- a/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.pyx
+++ b/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.pyx
@@ -1,6 +1,6 @@
 # cython: language_level=3, boundscheck=False, wraparound=False
-# distutils: extra_compile_args = ["-fopenmp", "-openmp"]
-# distutils: extra_link_args = ["-fopenmp", "-openmp"]
+# distutils: extra_compile_args = OPENMP_CCARGS
+# distutils: extra_link_args = OPENMP_CLARGS
 
 import numpy
 cimport numpy

--- a/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.pyx
+++ b/amulet_map_editor/api/opengl/mesh/level/chunk/chunk_builder_cy.pyx
@@ -1,6 +1,6 @@
 # cython: language_level=3, boundscheck=False, wraparound=False
-# distutils: extra_compile_args = -openmp
-# distutils: extra_link_args = -openmp
+# distutils: extra_compile_args = ["-fopenmp", "-openmp"]
+# distutils: extra_link_args = ["-fopenmp", "-openmp"]
 
 import numpy
 cimport numpy
@@ -14,6 +14,15 @@ cdef extern from "stdlib.h":
 from cpython cimport array
 import array
 from cython.parallel import prange
+
+cdef extern from *:
+    """
+    #ifdef _OPENMP
+    #pragma message ("openmp enabled")
+    #else
+    #pragma message ("openmp disabled")
+    #endif
+    """
 
 cdef float _brightness_step = 0.15
 cdef dict _brightness_multiplier = {

--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,4 @@
-from typing import List
+from typing import List, Tuple
 from setuptools import setup, find_packages
 from wheel.bdist_wheel import bdist_wheel
 from Cython.Build import cythonize
@@ -6,6 +6,8 @@ import glob
 import sys
 import numpy
 import subprocess
+import logging
+import re
 import versioneer
 
 
@@ -52,10 +54,62 @@ def freeze_requirements(packages: List[str]) -> List[str]:
     return packages
 
 
+GCCPattern = re.compile(r"gcc version (?P<major>\d+)\.(?P<minor>\d+)")
+ClangPattern = re.compile(r"clang(?:-|\s+version\s+)(?P<major>\d+)\.(?P<minor>\d+)")
+
+
+def get_openmp_args() -> Tuple[List[str], List[str], List[str], List[str]]:
+    # This has been lifted from here https://github.com/cython/cython/blob/606bd8cf235149c3be6876d0f5ae60032c8aab6c/runtests.py
+    import sysconfig
+    from distutils import ccompiler
+
+    def get_openmp_args_for(arg) -> Tuple[List[str], List[str]]:
+        """arg == 'CC' or 'CXX'"""
+        cc = (
+            sysconfig.get_config_var(arg) or ccompiler.get_default_compiler()
+        ).split()[0]
+        if cc == "msvc":
+            # Microsoft Visual C
+            return ["/openmp"], []
+        elif cc:
+            # Try GCC and Clang
+            try:
+                out = subprocess.check_output([cc, "-v"]).decode()
+            except ChildProcessError:
+                logging.exception(f"Could not resolve unknown compiler {cc}")
+            else:
+                gcc_match = GCCPattern.search(out)
+                if gcc_match:
+                    if (gcc_match.group("major"), gcc_match.group("minor")) >= (4, 2):
+                        return ["-fopenmp"], ["-fopenmp"]
+                    return [], []
+                clang_match = ClangPattern.search(out)
+                if clang_match:
+                    # if (clang_match.group("major"), clang_match.group("minor")) >= (3, 7):
+                    #     return ['-fopenmp'], ['-fopenmp']
+                    return [], []
+        # If all else fails disable openmp
+        return [], []
+
+    omp_ccargs, omp_clargs = get_openmp_args_for("CC")
+    omp_cppcargs, omp_cpplargs = get_openmp_args_for("CXX")
+
+    return omp_ccargs, omp_clargs, omp_cppcargs, omp_cpplargs
+
+
 # build cython extensions
 if next(glob.iglob("amulet_map_editor/**/*.pyx", recursive=True), None):
     # This throws an error if it does not match any files
-    ext = cythonize("amulet_map_editor/**/*.pyx")
+    omp_ccargs, omp_clargs, omp_cppcargs, omp_cpplargs = get_openmp_args()
+    ext = cythonize(
+        "amulet_map_editor/**/*.pyx",
+        aliases={
+            "OPENMP_CCARGS": omp_ccargs,
+            "OPENMP_CLARGS": omp_clargs,
+            "OPENMP_CPPCARGS": omp_cppcargs,
+            "OPENMP_CPPLARGS": omp_cpplargs,
+        },
+    )
 else:
     ext = ()
 

